src/m365/aad/commands/app/app-add.ts

```diff
@@ -104,7 +113,16 @@ class AadAppAddCommand extends GraphCommand {
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
     this
       .resolveApis(args, logger)
-      .then(apis => this.createAppRegistration(args, apis, logger))
+      .then(apis => {
+        return this.createAppRegistration(args, apis, logger)
+          .then(appInfo => {
+            if (args.options.grantAdminConsent && apis.length > 0) {
```

Comment:
If it's possible to add scopes using the manifest, this grant admin consent function call will be too early...

Reply:
Good catch!! 👏 Could you create a separate issue for this? I think it's better to discuss and handle it separately.

For now at least move the grandadminconsent function down so that both scopes added using the manifest and scopes added as options will be granted admin consent. It does not matter for this functionality whether the one overrides the other, just that they're both handled.

Reply:
Issue sumbitted here #3555 

Sounds good to me, all permissions FTW 😉